Makefile

```diff
@@ -17,12 +17,8 @@ install-common:
 	install -D -m 444 opensciencegrid.org.pub $(DESTDIR)/etc/cvmfs/keys/opensciencegrid.org
 
 install-debian: install-common
-	mkdir -p $(DESTDIR)/lib/systemd/system \
-	  $(DESTDIR)/lib/systemd/system/autofs.service.wants \
-	  $(DESTDIR)/usr/sbin
-	install -D -m 444 cvmfs-config-osg.service $(DESTDIR)/lib/systemd/system
-	ln -s ../cvmfs-config-osg.service $(DESTDIR)/lib/systemd/system/autofs.service.wants/cvmfs-config-osg.service
-	install -D -m 555 cvmfs-config-osgd $(DESTDIR)/usr/sbin
+	mkdir -p $(DESTDIR)/etc/auto.master.d
+	install -D -m 0644 cvmfs.autofs $(DESTDIR)/etc/auto.master.d
```

Comment:
Putting a cvmfs.autofs file here is breaking with the precedents set by cvmfs and by the cvmfs-config-* packages on Redhat.  For sure I don't think it belongs in a cvmfs-config-* package; if anything it would be in cvmfs itself, but Jakob has chosen to make this a separate manual configuration step, since not every installation wants autofs mounting.  I don't think we can assume that everybody on Debian wants it.  It should be part of the installation instructions instead.

Reply:
@DrDaveD This isn't a precedent so much as not having prioritized the adoption of a feature that I believe wasn't introduced until RHEL7. For example, a few months after the release of RHEL7, @jblomer [wrote a ticket to himself](https://sft.its.cern.ch/jira/browse/CVM-675) suggesting its adoption. @bbockelm also [requested its adoption](https://sft.its.cern.ch/jira/browse/CVM-1238) as has @rptaylor [here](https://sft.its.cern.ch/jira/browse/CVM-1426). The timing of the last report is suggestive that this PR was noticed or discussed on a another list?

I agree that the perfect place for this file is the `cvmfs` package itself. I try not to let the perfect be the enemy of the good, but perhaps @jblomer is willing to adopt it following this example?

I believe (strongly) that post-installation steps are an _intrinsic evil_ that increase support costs because, well, people don’t read things on a monitor very carefully. We have technical options for eliminating these post-installation steps and we should use them. I’m curious what @jblomer, @bbockelm, @rptaylor think about the idea and its proper location(s)?

Re: “autofs isn’t really required so we shouldn’t X/Y/Z”. `cvmfs` on RedHat does actually require the installation of `autofs`. On Debian, `cvmfs` “Recommends” `autofs` (which I’ve come to believe is the correct dependency) and the default behavior of apt-get is `—install-recommends`. So, in practice, the installation of `autofs` is automatic on all platforms and it’s perfectly reasonable to configure it.

The Debian-way, in contrast to RedHat, is to enable and start services when they are installed. `autofs` follows this approach, so it’s reasonable (on Debian) to also reload the autofs maps _if and only if_ the service is presently enabled. This is what the `postinst/postrm` scripts you note below do.

Reply:
@rptaylor can you point me to the mailing list?

Reply:
@tpdownes This is the mailing list: https://e-groups.cern.ch/e-groups/Egroup.do?egroupName=cvmfs-talk
Not sure if there is an online archive of emails.

Reply:
It is true that not everybody may want CVMFS mounted via autofs, but isn't that the reason for the optional cvmfs-auto-setup package existing? So those who want to configure autofs can do so by simply installing cvmfs-auto-setup. 
https://sft.its.cern.ch/jira/browse/CVM-675 is mainly about improving how cvmfs-auto-setup does that, while https://sft.its.cern.ch/jira/browse/CVM-1426 is mainly about removing redundant configuration steps so that cvmfs-auto-setup does no more than what is necessary.

To me it seems the change in this merge request should go in cvmfs-auto-setup, indeed not a cvmfs-config-* package.

Reply:
It may not be leaping out from the above... At no point have I made the recommendation to actively enable `autofs` via CVMFS or related packages. In fact, I disagree with the approach taken by `cvmfs-auto-setup`. No package should actively change the state of a different package's service. `postinst/postrm` scripts typically respect the configured state of the service packaged within. I can't see that I've ever seen one that changes another package's service state.

So, I think that approach could use some re-thinking.

Re: `auto.master.d/cvmfs.autofs`... The only situation I can see where the presence of this file on Debian or RedHat platforms where this file could possibly be bad is one in which `autofs` is running, but it is being avoided for CVMFS. Is this a real world issue?

If so, I agree that another package for `/etc/auto.master.d/cvmfs.autofs` is not entirely unreasonable. On Debian, I would suggest that this package be in a "Recommends" relationship with `cvmfs` so that User X installs it by default, but User Y can avoid it . I do not know if there is a similar relationship in RPMs.

Reply:
@jblomer's [ticket](https://sft.its.cern.ch/jira/browse/CVM-675) was for `cvmfs-config setup`, not for automatic configuration.  Brian's ticket didn't specify.  I think a separate package would be best.  A Recommends on Debian is a good idea.  If there's going to be a package, I think there should be a solution also for el6 which edits /etc/auto.master in a %post rule. 

@rptaylor Yes the cvmfs-auto-setup package enables autofs, but it does it by calling `cvmfs_config setup` which does too much, much more than enabling autofs.

@tpdownes In any case, please remove the autofs configuration from this pull request.

Reply:
I forgot to answer this:

> Re: `auto.master.d/cvmfs.autofs`... The only situation I can see where the
presence of this file on Debian or RedHat platforms where this file could
possibly be bad is one in which `autofs` is running, but it is being avoided for
CVMFS. Is this a real world issue?

One case is a cvmfs release manager (publishing) machine -- it can't have autofs for cvmfs enabled, but it could have autofs for mounting home directories or some such.

Reply:
@rptaylor @DrDaveD This is the entirety `cvmfs-auto-setup` does:
```
/usr/bin/cvmfs_config setup
service autofs status > /dev/null 2>&1
if [ $? -ne 0 ]; then
  service autofs start > /dev/null
fi
```
`cvmfs_config setup` enables and starts `autofs` but the script then goes ahead and checks whether it succeeded and tries again. So, not particularly transparent, confidence inspiring or line with my belief that a package shouldn't go around changing the status of services from other packages.

I'm happy to start a repo targeted at the following functionality:

* create /etc/auto.master.d/cvmfs.autofs
* reload the map if autofs is both enabled and running, otherwise `noop`

How hard would it be to get the `Recommends` addition to `cvmfs` itself? I would hope that could be a `2.4.x` thing.

Reply:
Please note: the cvmfs-auto-setup package was created on [request of one of the LHC experiments for a subset of their sites|http://cvmfs.readthedocs.io/en/stable/apx-rpms.html].  It's sole purpose was to give the administrators of this site an option to run `cvmfs_config setup` by installing a package.  I don't think it should have a wider audience or a broader purpose.

Reply:
@jblomer , what is the reason that some T3 sites require a package to specifically run 'cvmfs_config setup' ? If it is some custom need particular to a few sites, can't they create such an RPM themselves?
To me it seems like there is a bit of a disconnect here. On one hand, the cvmfs-auto-setup package was apparently created for a limited specific purpose for a small set of users and should not be generally used. On the other hand, it is supported and maintained in the main cernvm repository alongside the other official CVMFS packages, in a state that is useful and convenient (easy way to configure a node to use CVMFS via autofs) but somehow not recommended, intended or considered suitable for that purpose. (?)
Although using the cvmfs-auto-setup package may not be ideal (hence the ticket suggesting improvements) we - perhaps mistakenly - adopted and documented it as part of our standard Compute Canada client instructions. Was this ill-advised?

The CVMFS quick start guide includes the instruction to do 'cvmfs_config setup' so I don't see how the cvmfs-auto-setup RPM should be considered anything more or less than a convenient way of implementing the officially-recommended standard client configuration (whatever that is, and notwithstanding the fact that 'cvmfs_config setup' does more than generally necessary for autofs config).

Reply:
@rptaylor this package bears some patina. At the time, I was told that would make things easier for ATLAS, it was easy enough for us to do, and ever since it sits in the package repository.  If there is a strong opinion against, please bring it up at the workshop and/or we should clarify with ATLAS whether there is still a need. I'm happy to deprecate it if it is not needed anymore. Please let me know privately if you'd like me to forward the request to corresponding contact persons. For CC, it would probably be better not to use the cvmfs-auto-setup package because, as you pointed out, it is a bit of a misuse of RPMs.